packages/gestalt/src/Tag.js

```diff
@@ -67,44 +94,50 @@ export default function Tag(props: Props): Node {
     },
   );
 
-  const borderStyle =
-    disabled && !errorMessage
-      ? {
-          __style: { border: `solid 1px ${colorGray200}` },
-        }
-      : undefined;
+  const borderStyle = disabled && !hasIcon ? { border: `solid 1px ${colorGray200}` } : {};
+
+  if (!disabled && !onRemove) {
+    throw new Error('Non-disabled Tags must include an onRemove handler');
```

Comment:
Im ok,  but I feel this is an opaque error. If people set disable value dynamically, this error might get thrown after the engineer has landed the code. I think I would do it required and for disable eng would have to pass () => {}

Reply:
+1, I'd treat it like Checkbox where the handler is required, even if it could become disabled at some point

Reply:
That's a good point!